scheduler/replica_scheduler.go

```diff
@@ -91,8 +91,12 @@ func (rcs *ReplicaScheduler) chooseDiskCandidates(nodeInfo map[string]*longhorn.
 	}
 	// If there's no disk fit for replica on other nodes,
 	// try to schedule to node that has been scheduled replicas.
-	for _, node := range filterdNode {
-		diskCandidates = rcs.filterNodeDisksForReplica(node, replica, replicas)
+	// Avoid this if Replica Hard Anti-Affinity is enabled.
+	hardAntiAffinity, err := rcs.ds.GetSettingAsBool(types.SettingNameReplicaHardAntiAffinity)
+	if err != nil || !hardAntiAffinity {
```

Comment:
If err is returned, also log the error. This shouldn't happen

Reply:
Added a line to log the error.